src/hooks/wallets/useOnboard.ts

```diff
@@ -106,8 +114,8 @@ export const connectWallet = async (
 
   isConnecting = true
 
-  // On mobile, automatically choose WalletConnect
-  if (!options && isMobile()) {
+  // On mobile, automatically choose WalletConnect if there is no injected wallet
+  if (!options && isMobile() && !hasInjectedWallet()) {
```

Comment:
LGTM! Did you manage to test it with the coinbase wallet?

Reply:
Yes, it works nicely.